sklearn/model_selection/_validation.py

```diff
@@ -191,20 +197,30 @@ def cross_validate(estimator, X, y=None, groups=None, scoring=None, cv=None,
         delayed(_fit_and_score)(
             clone(estimator), X, y, scorers, train, test, verbose, None,
             fit_params, return_train_score=return_train_score,
-            return_times=True)
+            return_times=True, return_estimator=return_estimator)
         for train, test in cv.split(X, y, groups))
 
     if return_train_score:
-        train_scores, test_scores, fit_times, score_times = zip(*scores)
+        if return_estimator:
```

Comment:
There must be a neater way to write this without nested ifs. But alas I'm not sure what it is.